src/LineageTree/lineageTree.py

```diff
@@ -2166,12 +2181,250 @@ def norm(*args):
     #             d[i, j] = c[i, j] + min((d[i-1, j], d[i, j-1], d[i-1, j-1]))
     #     return d[-1, -1], d
 
+    def brute_force_register(self):
+        from scipy.spatial import ConvexHull
+
+        def kd_tree(target_time, source_time):
+            source_data = []
+            target_data = []
+            for cell in self.time_nodes[source_time]:
+                source_data.append(list(self.pos[cell]))
+            for cell in self.time_nodes[target_time]:
+                target_data.append(list(self.pos[cell]))
+            target_data = np.array(target_data)
+            source_data = np.array(source_data)
+            hull_source = ConvexHull(source_data)
+            hull_target = ConvexHull(target_data)
+            new_data_source = np.array(hull_source.points)
+            new_data_target = np.array(hull_target.points)
+
+            for simplice in hull_source.simplices:
+                new_data_source = np.append(
+                    new_data_source,
+                    generate_points_in_triangle(
```

Comment:
This does not seem to be imported?

Reply:
actually it is bellow, within the same function.

Maybe one thing you could do since all that is pretty big would be to create a new class that would be named `treeAlignment` for example and put all these messy functions in it so then you can document them however you want :D